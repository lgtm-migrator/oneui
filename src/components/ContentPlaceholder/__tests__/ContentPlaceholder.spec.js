import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import ContentPlaceholder from '../ContentPlaceholder';

describe('<ContentPlaceholder> that renders a content placeholder', () => {
    it('should render default placeholder', () => {
        const { container } = render(<ContentPlaceholder />);
        expect(container).toMatchSnapshot();
        expect(container).toHaveStyle(`
            animationDuration: '1s',
            height: null,
        `);
        expect(container.querySelector('.ContentPlaceholder__mask')).toHaveStyle(`
            width: '0%',
        `);
    });

    it('should affect styles when props are changed', () => {
        const { container } = render(
            <ContentPlaceholder duration={3} height={30} width={60} withoutMargin />
        );

        expect(container).toHaveStyle(`
            animationDuration: '3s',
            height: 30,
        `);

        expect(container.querySelector('.ContentPlaceholder__mask')).toHaveStyle(`
            width: '40%',
        `);

        expect(container.querySelector('.ContentPlaceholder--withoutMargin')).toBeVisible();
    });
});
