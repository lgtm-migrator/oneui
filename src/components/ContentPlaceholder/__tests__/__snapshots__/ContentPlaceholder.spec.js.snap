// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentPlaceholder> that renders a content placeholder should render default placeholder 1`] = `
<div>
  <div
    class="ContentPlaceholder"
    style="animation-duration: 1s;"
  >
    Â 
    <div
      class="ContentPlaceholder__mask"
      style="width: 0%;"
    />
  </div>
</div>
`;
